conans/util/files.py

```diff
@@ -386,55 +386,36 @@ def exception_message_safe(exc):
 
 
 def merge_directories(src, dst, excluded=None):
-    src = os.path.normpath(src)
-    dst = os.path.normpath(dst)
-    excluded = excluded or []
-    excluded = [os.path.normpath(entry) for entry in excluded]
-
-    def is_excluded(origin_path):
-        if origin_path == dst:
-            return True
-        rel_path = os.path.normpath(os.path.relpath(origin_path, src))
-        if rel_path in excluded:
-            return True
-        return False
-
-    def link_to_rel(pointer_src):
-        linkto = os.readlink(pointer_src)
-        if not os.path.isabs(linkto):
-            linkto = os.path.join(os.path.dirname(pointer_src), linkto)
-
-        # Check if it is outside the sources
-        out_of_source = os.path.relpath(linkto, os.path.realpath(src)).startswith(".")
-        if out_of_source:
-            # May warn about out of sources symlink
-            return
-
-        # Create the symlink
-        linkto_rel = os.path.relpath(linkto, os.path.dirname(pointer_src))
-        pointer_dst = os.path.normpath(os.path.join(dst, os.path.relpath(pointer_src, src)))
-        os.symlink(linkto_rel, pointer_dst)
+    from conans.client.file_copier import FileCopier
+    copier = FileCopier([src], dst)
+    copier(pattern="*", excludes=excluded)
+    return
 
-    for src_dir, dirs, files in walk(src, followlinks=True):
-        if is_excluded(src_dir):
-            dirs[:] = []
-            continue
 
-        if os.path.islink(src_dir):
-            link_to_rel(src_dir)
-            dirs[:] = []  # Do not enter subdirectories
-            continue
+def discarded_file(filename):
+    """
+    # The __conan pattern is to be prepared for the future, in case we want to manage our
+    own files that shouldn't be uploaded
+    """
+    return filename == ".DS_Store" or filename.startswith("__conan")
+
+
+def gather_files(folder):
+    file_dict = {}
+    symlinked_folders = {}
+    for root, dirs, files in walk(folder):
+        dirs[:] = [d for d in dirs if d != "__pycache__"]  # Avoid recursing pycache
```

Comment:
I wonder if it is a good moment to remove this hardcoded filter. In the same way we don't filter anymore ``.pyc`` or ``test_package``?

Reply:
What about the `discarded_file()` filters? (`*.pyo`, `*.pyc`, `.DS_Store`, `__conan`)

Reply:
all the Python .pyc, .pyo have been already removed from ``discarded_file()``, right? It only keeps the .DS_Store which is really annoying in OSX. That is the thing, it seems a bit assymetric to keep filtering ``__pycache__``, but allow ``.pyc``?

Reply:
Yeah, ok. I'll remove it. This is develop2, should we remove the `CONAN_KEEP_PYTHON_FILES` then?

Reply:
Sorry, I was looking at "develop" branch by mistake, sure, let's remove it.